### PR TITLE
Add hint for migration of servlets for RESTEasy Reactive

### PR DESCRIPTION
**Pull request by sponiro**
Based on [this](https://github.com/quarkusio/quarkus/discussions/34618) discussion I tried to create a small note about servlets to the guide "Migrating to RESTEasy Reactive". Please keep in mind that I am not a native speaker. 
When trying to build the documentation I got an error from YamlMetadataGeneratorTest. It forced me to temporarily add some attributes. I did not push these on purpose as I am not sure what they do or why they are missing on many documents.

### CODE REVIEW

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 1]
```diff
@@ -120,6 +120,18 @@ Quarkus uses smart defaults when determining the media type of Jakarta REST meth
 The difference between `quarkus-resteasy-reactive` and `quarkus-resteasy` is the use of `text/plain` as the default media type instead of `text/html`
 when the method returns a `String`.
 
+=== Servlets
+
+RESTEasy Reactive does **not** support servlets.
+If your project depends on servlets you have to migrate them.
+A common situation is when your application is developed with JAX-RS for a servlet-based environment.
```
Comment (geoand) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:127]: I don't think this sentence provides much value

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 2]
```diff
@@ -120,6 +120,18 @@ Quarkus uses smart defaults when determining the media type of Jakarta REST meth
...
+A servlet-based JAX-RS implementation must support injections of these types with the `@Context` annotation: `ServletConfig`, `ServletContext`, `HttpServletRequest` and `HttpServletResponse`.
+Since RESTEasy Reactive is not servlet-based these injections are not mandatory.
```
Comment (geoand) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:129]: I would add that not only are they not mandatory,  but their use will not work

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 3]
```diff
@@ -120,6 +120,18 @@ Quarkus uses smart defaults when determining the media type of Jakarta REST meth
...
+It is not always obvious that this will fail especially if you depend on an extension like `quarkus-undertow` which supplies the interfaces.
```
Comment (geoand) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:130]: Maybe give an example here?

**Review by geoand — COMMENTED** [review]
Thanks for this. I added a few notes

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc (hunk 2 above)
Comment (sponiro) [reply docs/src/main/asciidoc/resteasy-reactive-migration.adoc]: I adjusted the sentence.

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc (hunk 1 above)
Comment (sponiro) [reply docs/src/main/asciidoc/resteasy-reactive-migration.adoc]: I removed it.

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc (hunk 3 above)
Comment (sponiro) [reply docs/src/main/asciidoc/resteasy-reactive-migration.adoc:130]: I added an example.

**Comment by geoand** [comment]
Can you please squash into a single commit?

**Comment by sponiro** [comment]
Commits are squashed

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**